rdb/Cursor.php

```diff
@@ -17,6 +17,7 @@ class Cursor implements Iterator
     private $currentData;
     private $currentSize;
     private $currentIndex;
+    private $resetIndex = 0;
```

Comment:
Maybe instead of `$resetIndex`, it would be nicer to have `$totalIndex`. We would initialize it to 0, and increment it in `next()`.
It would simplify the code in `key()`.